genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/JobKickoffTask.java

```diff
@@ -203,15 +203,17 @@ protected synchronized void createUser(final String user, final String group) th
         } catch (final IOException ioe) {
             log.debug("User does not exist. Creating it now.");
 
-            // Create the group for the user.
-            if (group != null) {
+            // Create the group for the user if its not the same as the user.
+            if (group != null && !group.equals(user) && StringUtils.isNotBlank(group)) {
+                log.debug("Group and User are different so creating group now.");
                 final CommandLine groupCreateCommandLine = new CommandLine("sudo");
                 groupCreateCommandLine.addArgument("groupadd");
                 groupCreateCommandLine.addArgument(group);
 
                 // We create the group and ignore the error as it will fail if group already exists.
                 // If the failure is due to some other reason, then user creation will fail and we catch that.
                 try {
+                    log.debug("Running command [" + groupCreateCommandLine.toString() + "]");
```

Comment:
Can you change this to Running command to create group so that it's clearer when we're grepping for it?